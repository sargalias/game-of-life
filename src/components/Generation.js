import React from 'react';

const Generation = () => (
  <div id="generation">
    Generation: <span id="generation-number">0</span>
  </div>
);

export default Generation;
