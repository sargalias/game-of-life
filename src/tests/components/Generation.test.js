import React from 'react';
import { shallow } from 'enzyme';
import Generation from '../../components/Generation';

test('should render Generation correctly', () => {
  const wrapper = shallow(<Generation />);
  expect(wrapper).toMatchSnapshot();
});

test('Generation should contain correct initial text', () => {
  const initialText = "Generation: 0";
  const wrapper = shallow(<Generation />);
  expect(wrapper.text()).toBe(initialText);
});
