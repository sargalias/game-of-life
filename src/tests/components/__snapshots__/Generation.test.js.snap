// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Generation correctly 1`] = `
<div
  id="generation"
>
  Generation: 
  <span
    id="generation-number"
  >
    0
  </span>
</div>
`;
